### DISCUSSION TITLE
(golang) implement java string.hash

### DISCUSSION THREAD

**Discussion by zenyui** [chief-of-state/chief-of-state#196] [Ideas]
we need our partitioner in golang to operate the same as the scala/java version, which relies on the [String.hashCode](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#hashCode--) method

should be straight forward to implement/test:
```
s[0]*31^(n-1) + s[1]*31^(n-2) + ... + s[n-1]
```

**Comment by Tochemey** [comment]
The implementation is done.
